src/main/java/in/projecteka/consentmanager/link/discovery/Discovery.java

```diff
@@ -85,18 +85,16 @@
                 .switchIfEmpty(Mono.error(ClientError.requestAlreadyExists()))
                 .flatMap(val ->
                         userWith(userName)
-                                .zipWith(gatewaySystemUrl())
-                                .flatMap(userAndGateway ->
-                                        discoveryServiceClient.requestPatientFor(
-                                                requestFor(userAndGateway.getT1(), transactionId, unverifiedIdentifiers),
-                                                userAndGateway.getT2(),
-                                                providerId))
-                                .zipWith(Mono.delay(Duration.ofSeconds(getExpectedFlowResponseDuration())))
-                                .flatMap(tuple2 ->
-                                        discoveryResults.get(transactionId.toString())
-                                                .switchIfEmpty(Mono.error(ClientError.gatewayTimeOut()))
-                                                .flatMap(dr -> resultFromHIP(dr))
-                ))
+                                .flatMap(user -> discoveryServiceClient.requestPatientFor(
+                                        requestFor(user, transactionId, unverifiedIdentifiers),
+                                        gatewaySystemUrl(),
+                                        providerId)
+                                        .zipWith(Mono.delay(Duration.ofSeconds(getExpectedFlowResponseDuration())))
```

Comment:
If the intent is to have discovery call cancelled when it runs beyond `getExpectedFlowResponseDuration`, I think it would be necessary to signal that from the delay Mono with say a `empty()` as in 
`zipWith(Mono.delay(Duration.ofSeconds(getExpectedFlowResponseDuration())).then(Mono.empty()))`.

Reply:
actually, post delay, we check in the redis cache for the value. only if the value is empty do we return error. So can't return empty mono

Reply:
not sure if am getting this right, are we concerned with potential delay of the discovery call in itself? In the sense, do we need to bother about discoveryclient call itself taking more than SLA time?

Reply:
Also maybe not for current scope, but we may need to look at https://redis.io/topics/notifications to avoid waiting for a maximum period.

Reply:
yes. we have to put that for logging for all request-response cycles. each server needs to track whether a request is received in time or not at all.